rclpy/rclpy/timer.py

```diff
@@ -131,7 +131,6 @@ def _postsleep(self):
                 self._event.clear()
         if self._is_shutdown:
             self.destroy()
-            raise ROSInterruptException()
```

Comment:
I don't think this code path is part of a timer being canceled. This exception is raised when `self._is_shutdown` is set, which happens when the context is shutdown.